hasktorch/src/Torch/Functions.hs

```diff
@@ -160,3 +160,6 @@ maxPool2d input (kh, kw) (dh, dw) (ph, pw) = unsafePerformIO $
 
 logSoftmax :: Tensor -> Int -> Tensor
 logSoftmax input dim = unsafePerformIO $ (cast3 ATen.log_softmax_tls) input dim (dtype input)
+
+diag :: Tensor -> Int -> Tensor
+diag t padding = unsafePerformIO $ (cast2 ATen.tensor_diag_l) t padding
```

Comment:
I think the third argument is not `padding` but it's the index of the diagonal to select

Reply:
![image](https://user-images.githubusercontent.com/20875313/60849079-331aeb00-a1b7-11e9-81ff-cd6f91e4e4b3.png)

i wasn't sure at first but this was the behaviour when i tried it.

Reply:
nvm iswym, padding is what happens when selecting an index..